provisioning/provisioning-device-client/src/main/java/com/microsoft/azure/sdk/iot/provisioning/device/internal/exceptions/ProvisioningDeviceClientExceptionManager.java

```diff
@@ -18,38 +18,30 @@ public static void verifyHttpResponse(HttpResponse httpResponse) throws Provisio
     {
         int responseStatus = httpResponse.getStatus();
 
-        String errorMessage = ErrorMessageParser.bestErrorMessage(new String(httpResponse.getErrorReason(), StandardCharsets.UTF_8));
```

Comment:
There is no reason for us to parse the error message considering it will just be a string to the exception we give it to. Parsing this error message may even filter out important information from the service, so we're better off just giving the full error payload string to the user